javascript/README.md

```diff
@@ -0,0 +1,156 @@
+                        The number of times literals should be simplified. 
+                        Runs until fixed point by default.
+  -c , --keep_comments 
+                        To keep all comments
+
+Required arguments:
+  -s SOURCE, --source SOURCE
+                        Path of input file for refactoring
+  -f FLAG, --flag FLAG  Name of the stale flag
+  -p PROPERTIES, --properties PROPERTIES
+                        Path of configuration file for Piranha
+```
+
+### Example 
+
+Consider the following properties file `config/properties.json`.
+
+```json
+{
+  "methodProperties": [
+    {
+      "methodName": "isToggleDisabled",
+      "flagType": "control",
+      "argumentIndex": 0
+    },
+    {
+      "methodName": "isFlagTreated",
+      "flagType": "treated",
+      "argumentIndex": 0
+    },
+    {
+      "methodName": "isToggleEnabled",
+      "flagType": "treated",
+      "argumentIndex": 0
+    }
+  ]
+}
+```
+
+This specifies a list of flag APIs each of which return a boolean value. It is assumed that a method returns `true` iff the `flagType` specified in the properties file matches the flag behaviour specified at runtime. For example, if the flag behavior is `treated`, `isFlagTreated(flag)` will return `true` and `isToggleDisabled(flag)` will return `false`.
+
+The code in `examples/sample.js` contains the stale flag `featureFlag`
+
+```javascript
+// Simple flag cleanup in conditional
+if(isFlagTreated(featureFlag)) {
+    f1();
+ } else {
+    f2();
+ }
+ 
+ // Assignment cleanup
+ var a = isToggleDisabled(featureFlag);
+ if(a) {
+    f1(); 
+ } else {
+    f2();
+ }
+ 
+ // function cleanup
+ function b() {
+     return isFlagTreated(featureFlag);
+ }
+ 
+ if(b() || f1()) {
+    f1()   
+ } else {
+    f2()   
+ }
+ 
+ // Complex cleanup
+ var c = isToggleDisabled(featureFlag) ? f1() : isToggleDisabled(featureFlag) ? f2() : isFlagTreated(featureFlag) ? f3() : f4();
+```
+When this file is run through Piranha with the command 
+
+```
+node src/piranha.js -s examples/sample.js -p config/properties.json --treated -f featureFlag
+```
+
+`sample.js` is modified in-place to yield
+
+```javascript
+// Simple flag cleanup in conditional
+f1();
+
+// Assignment cleanup
+f2();
+f1()
+
+// Complex cleanup
+var c = f3();
+```
+
+Note that code-style and comments are preserved wherever possible by default.
+
+### Running tests
+
+To run unit and integration tests invoke
+
+```
+npm test
+```
+
+See the suite of tests in `test/test_refactor.js` and `test/test_integration.js` for more detail on how Piranha works and what kind of refactoring it can handle. 
+
+### Running piranha as a command
+
+It can be convenient to run piranha as a shell command from any location. This can be done by creating an alias.
+
+From the *piranha/javascript* directory, run the following shell commands (for the bash shell).
+
+```
+printf "alias piranha='node `realpath src/piranha.js`'" >> ~/.bashrc
```

Comment:
maybe worth naming this alias `piranhajs`?

Reply:
Done 48929a4f680bf1a451432976cfaac1481f6ccc3e